common/reconciliation/invariants/concreteExecutionExists.go

```diff
@@ -59,6 +60,11 @@ func (c *concreteExecutionExists) Check(execution interface{}) common.CheckResul
 			InvariantType:   c.InvariantType(),
 		}
 	}
```

Comment:
Not related to your change but is the check from L57-62 correct? Even if the state for current execution is closed, as long as corresponding concrete execution doesn't exist, it's still a corruption I think?

Reply:
Yes. You are correct